test/user.test.js

```diff
@@ -879,42 +882,53 @@ describe('User', function() {
         assert(err && !/verified/.test(err.message),
           'expecting "login failed" error message, received: "' + err.message + '"');
         assert.equal(err.code, 'LOGIN_FAILED');
-
+        assert.equal(err.userId, undefined);
         done();
       });
     });
 
-    it('Login a user by without email verification', function(done) {
+    it('does not login a user with unverified email but provides userId', function(done) {
       User.login(validCredentials, function(err, accessToken) {
-        assert(err);
-        assert.equal(err.code, 'LOGIN_FAILED_EMAIL_NOT_VERIFIED');
-
+        err = Object.assign({}, err);
+        expect(err).to.eql({
+          statusCode: 401,
+          code: 'LOGIN_FAILED_EMAIL_NOT_VERIFIED',
+          details: {
+            userId: validCredentialsUser.pk,
+          },
+        });
         done();
       });
     });
 
-    it('Login a user by without email verification - promise variant', function(done) {
-      User.login(validCredentials)
-        .then(function(err, accessToken) {
-          done();
-        })
-        .catch(function(err) {
-          assert(err);
-          assert.equal(err.code, 'LOGIN_FAILED_EMAIL_NOT_VERIFIED');
-
-          done();
-        });
+    it('does not login a user with unverified email but provides userId ' +
+      '- promise variant', function() {
```

Comment:
IMO, there is no need to test both callback and promise variants - there is no promise/callback specific code in path triggered by this test case, therefore it's enough to test a single variant (preferably the promise variant).